activerecord/lib/active_record/transactions.rb

```diff
@@ -340,7 +340,7 @@ def before_committed! # :nodoc:
     # Ensure that it is not called if the object was never persisted (failed create),
     # but call it after the commit of a destroyed object.
     def committed!(should_run_callbacks: true) #:nodoc:
-      if should_run_callbacks && destroyed? || persisted?
+      if should_run_callbacks && (destroyed? || persisted?)
```

Comment:
This bug was accidentally introduced at https://github.com/rails/rails/pull/16537#discussion_r16446313.